presto-orc/src/main/java/com/facebook/presto/orc/metadata/DwrfMetadataReader.java

```diff
@@ -200,20 +200,29 @@ private static DwrfSequenceEncoding toSequenceEncoding(OrcType type, DwrfProto.C
                         columnEncoding.getDictionarySize()));
     }
 
-    private static Optional<SortedMap<Integer, DwrfSequenceEncoding>> toAdditionalSequenceEncodings(List<DwrfProto.ColumnEncoding> columnEncodings, OrcType type)
+    private static ColumnEncoding toColumnEncoding(OrcType type, List<DwrfProto.ColumnEncoding> columnEncodings)
     {
-        if (columnEncodings.size() == 1) {
-            return Optional.empty();
+        DwrfProto.ColumnEncoding sequence0 = null;
+        ImmutableSortedMap.Builder<Integer, DwrfSequenceEncoding> builder = ImmutableSortedMap.naturalOrder();
+        for (DwrfProto.ColumnEncoding columnEncoding : columnEncodings) {
+            if (columnEncoding.getSequence() == 0) {
```

Comment:
the columnEncoding sequence 0 if present, should be the first value, so the if check can be moved outside the for loop.

sequence0 can be assigned to the one if one exists, or a dummy DWRF_DIRECT encoding can be created, so that the below code does not need to switch on sequence0 != null.

Reply:
even though sequence 0 is indeed the first one in `columnEncodings` in current implementation, the spec doesn't guarantee order